### DISCUSSION TITLE
Which taskserver sync mechanism to build upon?

### DISCUSSION THREAD

**Discussion by a-t-0** [GothenburgBitFactory/taskwarrior#3106] [Q&A] [answered]
## Context
A while ago, I wrote [a poor automation of the taskserver installation](https://github.com/HiveMinds/tw-install), such that it syncs with taskwarrior. Now I've automated (somewhat decent) [self-signed SSL certificates for https onion domains](https://github.com/HiveMinds/SSL4Tor). I would like to apply these auto-generated SSL certificates to taskserver, to make the self-hosted taskserver (for taskwarrior and timewarrior) run from any place with internet/tor access (within a single command).

Given the wide scope of development (e.g. with taskchampion ([related thread](https://github.com/GothenburgBitFactory/taskwarrior/discussions/2967#discussion-4468350)), and [Task Git Sync](https://github.com/amracks/tgs) etc.), I would like to verify my efforts on automating this self-hosted taskwarrior synchronisation mechanism over tor, are aimed at a non-outdated sync protocol. Therefore, I would like to ask:

## Question
Is the [current taskserver sync mechanism](https://github.com/GothenburgBitFactory/taskserver) still the recommended taskserver mechanism, or would it be advisable to look at a different, still-in-development sync mechanism?

**Comment by djmitche** [comment, marked as answer]
I am working toward dropping support for taskserver and the existing sync functionality in https://github.com/GothenburgBitFactory/taskwarrior/pull/3094. I can't tell the future, but my hope is that 3.0 will use Taskchampion.

One of the things that needs doing before that time is "productionizing" the taskchampion-sync-server. It's currently not very 12-factor compatible (takes config as command-line arguments) and allows creation of new clients without any other kind of authentication. It'd be great to have some perspective on that, leading to a set of issues to fix. Some of those will require Rust work, but some will be more operational in nature. Are you interested?

**Comment by a-t-0** [comment]
Thank you for your clear and quick response, and for acquainting me with twelve-factor compatibility. I would love to contribute, and get hands-on with rust, however, I am currently focussed on finishing my studies and finding an income. So if it would be possible to receive some funding for the work, I would be happy to work on "productionizing" the taskchampion-sync-server.

**Reply by djmitche** [reply]
Sounds great!
